File: Week 6 Python/sentimental-mario-less/mario.py
# TODO

def main():
    size = get_size()
    print_obstacle(size)


def get_size():
    # wait for correct user input
    while True:
        try:
            n = int(input("Height: "))
            if n in range(1, 9):
                return n
        except ValueError:
            pass


def print_obstacle(size):
    # declare counter
    count = size
    # loop and print the correct shape
    for i in range(size):
        for j in range(size):
            if j < count - 1:
                print(" ", end="")
            elif j >= count:
                print("#", end="")
            else:
                print("#", end="")
                count -= 1
        print()


main()

File: Week 1 C/mario-less/mario.c
#include "cs50.h"
#include "stdio.h"

int get_size(void);
void print_obstacle(int size);

int main(void)
{
  int size = get_size();
  print_obstacle(size);
}

int get_size(void)
{
  int n;
  do
  {
    n = get_int("Height: ");
  } while (n > 8 || n < 1);
  return n;
}

void print_obstacle(int size)
{
  int count = size;
  for (int i = 0; i < size; i++)
  {
    for (int j = 0; j < size; j++)
    {
      if (j < count - 1)
      {
        printf(" ");
      }
      else if (j >= count)
      {
        printf("#");
      }
      else
      {
        printf("#");
        count--;
      }
    }
    printf("\n");
  }
}